README.rst

```diff
@@ -87,155 +129,156 @@ must be present and contain a dict with Zyte API parameters:
+``zyte_api_automap`` key in
+`Request.meta <https://docs.scrapy.org/en/latest/topics/request-response.html#scrapy.http.Request.meta>`_
+to ``True``. See also **Using transparent mode** below.
 
-Setting default parameters
---------------------------
-Often the same configuration needs to be used for all Zyte API requests.
-For example, all requests may need to set the same geolocation, or
-the spider only uses ``browserHtml`` requests.
+Automated parameter mapping chooses Zyte API parameters as follows by default:
 
-To set the default parameters for Zyte API enabled requests, you can set the
-following in the ``settings.py`` file or `any other settings within Scrapy
-<https://docs.scrapy.org/en/latest/topics/settings.html#populating-the-settings>`_:
+-   ``httpResponseBody`` and ``httpResponseHeaders`` are set to ``True``.
 
-.. code-block:: python
+-   ``Request.url`` becomes ``url``, same as in requests with manually-defined
+    parameters.
 
-    ZYTE_API_DEFAULT_PARAMS = {
-        "browserHtml": True,
-        "geolocation": "US",
-    }
+-   If ``Request.method`` is something other than ``"GET"``, it becomes
+    ``httpRequestMethod``.
 
+-   ``Request.headers`` become ``customHttpRequestHeaders``.
 
-``ZYTE_API_DEFAULT_PARAMS`` works if the ``zyte_api``
-key in `Request.meta <https://docs.scrapy.org/en/latest/topics/request-response.html#scrapy.http.Request.meta>`_
-is set, i.e. having ``ZYTE_API_DEFAULT_PARAMS`` doesn't make all requests
-to go through Zyte API. Parameters in ``ZYTE_API_DEFAULT_PARAMS`` are merged
-with parameters set via the ``zyte_api`` meta key, with the values in meta
-taking priority.
+-   ``Request.body`` becomes ``httpRequestBody``.
 
-.. code-block:: python
+Instead of setting ``zyte_api_automap`` to ``True``, you may set it to a
+``dict`` of Zyte API parameters to extend or override choices made by automated
+parameter mapping. Some parameters modify the result of automated parameter
+mapping as a side effect:
 
-    import scrapy
+-   Setting ``browserHtml`` or ``screenshot`` to ``True`` unsets
+    ``httpResponseBody``, and makes ``Request.headers`` become
+    ``requestHeaders`` instead of ``customHttpRequestHeaders``.
 
+-   Setting ``screenshot`` to ``True`` without also setting ``browserHtml`` to
+    ``True`` unsets ``httpResponseHeaders``.
 
-    class SampleQuotesSpider(scrapy.Spider):
-        name = "sample_quotes"
+When mapping headers, unsupported headers are excluded from the mapping. If
+Zyte API header support changes in the future, and you cannot upgrade to a
+version of scrapy-zyte-api that includes such changes, use the following
+settings to change which headers are mapped and how they are mapped:
 
-        custom_settings = {
-            "ZYTE_API_DEFAULT_PARAMS": {
-                "geolocation": "US",  # You can set any Geolocation region you want.
-            }
-        }
+-   ``ZYTE_API_UNSUPPORTED_HEADERS`` determines headers that *cannot* be mapped
```

Comment:
I wonder if there are other reasons to skip the headers, not only that API doesn't support them. E.g. user-agent, or cookies, not sure. If there are such reasons, it could make sense to rename the option to something like ZYTE_API_SKIP_HEADERS, as the headers may be supported, but user still may want to avoid sending them.